### PR TITLE
[#139556943] try to make group_by work for units

### PR DESCRIPTION
**Pull request by jessicarobins**

### CODE REVIEW

**Comment by jessicarobins** [comment]
this actually works i just tested it

File: lib/unit/base.rb [hunk 1]
```diff
@@ -19,6 +19,10 @@ def scale_hash
       self.class.scale_hash
     end
 
+    def hash
+      scalar.hash ^ uom.hash
```
Comment (doyle) [inline lib/unit/base.rb]: Does
```
Unit.parse('1000 mcg').hash == Unit.parse('1 mg').hash
```

File: test/unit_test.rb [hunk 2]
```diff
@@ -1,6 +1,38 @@
...
+  end
+
+  context "different but compatible units" do
+    setup do
+      @u1 = Unit.parse('100 mcg')
+      @u2 = Unit.parse('0.1 mg')
+    end
+
+    should "be equal" do
+      assert @u1.eql? @u2
```
Comment (doyle) [inline test/unit_test.rb]: Can you check that the `hash` is equal too?

**Comment by doyle** [comment]
Merged 🐎

**Pull request merged**